### PR TITLE
feat: Add ariaControls prop to RadioGroup

### PR DESCRIPTION
**Pull request by fa7ad**
### Description
This PR adds the `ariaControls` prop to RadioGroup component.
This should allow users to implement a progressive disclosure pattern AND make it accessible.

~~NOTE: The API documentation needs to reviewed and potentially changed; hold off on merging just yet.~~ _approved_

Related links, issue #, if available: n/a

### How has this been tested?
* Updated existing Unit Tests
* Created a dev page that uses the prop to implement this pattern.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pages/radio-group/progressive-disclosure.page.tsx
Comment (just-boris) [inline pages/radio-group/progressive-disclosure.page.tsx:88]: Should it be `display:none`, `aria-hidden=true` or just a conditional rendering? What happens if aria controls resolves to a non-existing id?

File: pages/radio-group/progressive-disclosure.page.tsx
Comment (just-boris) [inline pages/radio-group/progressive-disclosure.page.tsx:49]: Impressive demo, but a little bit over-engineered for this case. It is extra complexity when somebody tries to read this demo and apply it for their case

1. Remove `useReducer`, plain state will do
2. Remove extra options from radio group, we only need to show case how to show/hide an extra control


Basically, this demo should show a minimal reproducible example of `ariaControls` use-case.

File: src/radio-group/interfaces.ts
Comment (just-boris) [inline src/radio-group/interfaces.ts:44]: How about this

```suggestion
   * Adds `aria-controls` attribute to the radio group. If radio group selection controls presence of another form field, use this property to provide an id the controlled field
```

**Review by just-boris — APPROVED** [review]

File: pages/radio-group/progressive-disclosure.page.tsx
Comment (just-boris) [reply pages/radio-group/progressive-disclosure.page.tsx:88]: Can we do conditional rendering? (I think this should be recommended, for example to allow lazy-loading options for conditional select). If not, what is the alternative?

Comment (timogasda) [reply pages/radio-group/progressive-disclosure.page.tsx:88]: > What happens if aria controls resolves to a non-existing id?

Certain a11y checkers will throw a violation 👀

Comment (fa7ad) [reply pages/radio-group/progressive-disclosure.page.tsx:88]: Exactly, the referenced ID is not in the document; some checkers will throw a violation. <sup>[example](https://rocketvalidator.com/html-validation/the-aria-controls-attribute-must-point-to-an-element-in-the-same-document)</sup>

Comment (just-boris) [reply pages/radio-group/progressive-disclosure.page.tsx:88]: How is it different if hidden elements are not in the a11y tree anyway?

Comment (just-boris) [reply pages/radio-group/progressive-disclosure.page.tsx:88]: P.S. if you have an expert approval on this pattern, I can approve the PR after addressing other comments.

But my general software development experience suggests me that something is does not match up here

Comment (fa7ad) [reply pages/radio-group/progressive-disclosure.page.tsx:88]: `aria-controls` is not limited to visible elements; at least as far as the spec is concerned.

https://www.w3.org/TR/wai-aria-1.1/#aria-controls
For example, Tab content visibility is given as a valid use of aria-controls

In the discussion for this issue, they make the distinction between non-existent and non-visible elements.
https://github.com/w3c/aria/issues/531

File: src/radio-group/interfaces.ts
Comment (just-boris) [reply src/radio-group/interfaces.ts:44]: Still would like to hear what you think about this suggestion

**Review by just-boris — DISMISSED** [review]

File: src/radio-group/interfaces.ts
Comment (fa7ad) [reply src/radio-group/interfaces.ts:44]: I'm generally open to the change, but not so sure about the suggestion.

I'm not sure if this part is syntactically correct: `to provide an id the controlled field`.

How about this instead?

```js
   /**
    * Adds `aria-controls` attribute to the radio group. 
    * If the radio group controls the visibility of any secondary content (like another FormField), provide an ID that refers to the secondary content.
    */

```

Comment (fa7ad) [reply src/radio-group/interfaces.ts]: Updated API doc following offline discussion

**Review by just-boris — APPROVED** [review]

**Pull request merged**
